drivers/mmcsd/mmcsd_sdio.c

```diff
@@ -1625,6 +1653,23 @@ static ssize_t mmcsd_readmultiple(FAR struct mmcsd_state_s *priv,
       SDIO_RECVSETUP(priv->dev, buffer, nbytes);
     }
 
+#ifdef CONFIG_MMCSD_MMCSUPPORT
+  ret = mmcsd_setblockcount(priv, nblocks);
+  if (ret != OK)
+    {
+      return ret;
+    }
+#else
+  if (priv->cmd23support)
```

Comment:
You only need this because priv->cmd23support is 1 for MMC
```
  if (priv->cmd23support)
    {
      ret = mmcsd_setblockcount(priv, nblocks);
      if (ret != OK)
        {
          return ret;
        }
    }
```

Reply:
@ldube   The reasons are the same as above